Source/Tests/LinuxSyscalls/FileManagement.cpp

```diff
@@ -138,13 +138,45 @@ void FileManager::LoadThunkDatabase(bool Global) {
           }
         }
         else if (strcmp(ItemName, "Overlay") == 0) {
+
+          auto AddWithReplacement = [DBObject](json_t const* Value) {
+            constexpr static std::array<const char*, 4> Prefixes = {
+              "/usr/lib",
+              "/usr/local/lib",
+              "/lib",
+              "/usr/lib/pressure-vessel/overrides/lib",
+            };
+
+            auto ReplacePrefix = [](const std::string &String, const char *Prefix, const char *NewPrefix) -> std::pair<bool, std::string> {
+              auto it = String.find(Prefix);
+              if (it != String.size()) {
+                size_t SizeOfOldPrefix = strlen(Prefix);
+                std::string Replacement = String;
+                Replacement.replace(it, SizeOfOldPrefix, NewPrefix);
+                return std::make_pair(true, Replacement);
+              }
+              else {
+                return std::make_pair(false, String);
+              }
+            };
+
+            bool Replace = true;
+            size_t i = 0;
+            do {
+              std::string NonModifiedLibraryItem {static_cast<const char*>(json_getValue(Value))};
+              auto Replacement = ReplacePrefix(NonModifiedLibraryItem, "@PREFIX@", Prefixes[i]);
+              DBObject->second.Overlays.emplace_back(Replacement.second);
+              Replace = Replacement.first;
+            } while (Replace && ++i != Prefixes.size());
```

Comment:
```suggestion
            std::string NonModifiedLibraryItem {static_cast<const char*>(json_getValue(Value))};
            for (auto& prefix : Prefixes[i]) {
              auto [has_prefix, modified_library_item] = ReplacePrefix(NonModifiedLibraryItem, "@PREFIX@", Prefixes[i]);
              DBObject->second.Overlays.emplace_back(std::move(modified_library_item));
              if (!has_prefix) {
                  break;
              }
            }
```

Instead of the `while (Replace)`/`if (!has_prefix)` checks, wouldn't a hard error be more appropriate? Seems like *not* starting any of the paths in the thunks ini with `@PREFIX@` would likely yield results that are unexpected otherwise.

Reply:
Not supporting `@PREFIX@` on the path is important for quick hacking of absolute paths for testing. Might also be necessary for replacing paths that don't match the typical `/usr` prefix.

Reply:
Sounds good. I slightly misread the code before, so this shouldn't cause any unexpected results after all.